src/ethereum/ethEngine.js

```diff
@@ -96,7 +104,11 @@ export class EthereumEngine extends CurrencyEngine {
   async fetchGetEtherscan (server: string, cmd: string) {
     const { etherscanApiKey } = this.initOptions
     let apiKey = ''
-    if (etherscanApiKey && etherscanApiKey.length > 5) {
+    if (
+      etherscanApiKey &&
+      etherscanApiKey.length > 5 &&
+      server.includes('etherscan')
```

Comment:
If this function is used for more than just Etherscan, should it have a different name?

Reply:
Ah, I get it. It's the Etherscan API, but not necessarily hosted by Etherscan the company. Seems fine, then.